exchange/handlers.go

```diff
@@ -67,7 +67,7 @@ type DeviceHandler func(id string, token string) (*Device, error)
 
 func GetHTTPDeviceHandler(ec ExchangeContext) DeviceHandler {
 	return func(id string, token string) (*Device, error) {
-		return GetExchangeDevice(ec.GetHTTPFactory(), ec.GetExchangeId(), ec.GetExchangeToken(), ec.GetExchangeURL())
+		return GetExchangeDevice(ec.GetHTTPFactory(), id, token, ec.GetExchangeURL())
```

Comment:
On node it does not make a difference. On the agbot, if you try to query a node, the old code will query the agbot instead of the node that is given from the input. I forgot where I had to use it.